src/Staking.sol

```diff
@@ -89,84 +91,99 @@ contract StakeRegistry is AccessControl, Pausable {
-        uint256 withDrawLimit = (amount > stake.stakeAmount) ? stake.stakeAmount : amount;
+        uint256 withDrawLimit = (_amount > stake.stakeAmount) ? stake.stakeAmount : _amount;
         stake.stakeAmount -= withDrawLimit;
 
         if (stake.stakeAmount == 0) {
-            delete stakes[overlay];
+            delete stakes[_owner];
         } else {
-            stakes[overlay].lastUpdatedBlockNumber = block.number;
+            stakes[_owner].lastUpdatedBlockNumber = block.number;
         }
 
         if (!ERC20(bzzToken).transfer(msg.sender, withDrawLimit)) revert TransferFailed();
     }
 
     /**
      * @dev Freeze an existing stake, can only be called by the redistributor
-     * @param overlay the overlay selected
-     * @param time penalty length in blocknumbers
+     * @param _owner the addres selected
+     * @param _time penalty length in blocknumbers
      */
-    function freezeDeposit(bytes32 overlay, uint256 time) external {
+    function freezeDeposit(address _owner, uint256 _time) external {
         if (!hasRole(REDISTRIBUTOR_ROLE, msg.sender)) revert OnlyRedistributor();
 
-        if (stakes[overlay].isValue) {
-            stakes[overlay].lastUpdatedBlockNumber = block.number + time;
-            emit StakeFrozen(overlay, time);
+        if (stakes[_owner].isValue) {
+            stakes[_owner].lastUpdatedBlockNumber = block.number + _time;
+            emit StakeFrozen(_owner, stakes[_owner].overlay, _time);
         }
     }
 
     /**
      * @dev Slash an existing stake, can only be called by the `redistributor`
-     * @param overlay the overlay selected
-     * @param amount the amount to be slashed
+     * @param _owner the _owner adress selected
+     * @param _amount the amount to be slashed
      */
-    function slashDeposit(bytes32 overlay, uint256 amount) external {
+    function slashDeposit(address _owner, uint256 _amount) external {
         if (!hasRole(REDISTRIBUTOR_ROLE, msg.sender)) revert OnlyRedistributor();
 
-        if (stakes[overlay].isValue) {
-            if (stakes[overlay].stakeAmount > amount) {
-                stakes[overlay].stakeAmount -= amount;
-                stakes[overlay].lastUpdatedBlockNumber = block.number;
+        if (stakes[_owner].isValue) {
+            if (stakes[_owner].stakeAmount > _amount) {
+                stakes[_owner].stakeAmount -= _amount;
+                stakes[_owner].lastUpdatedBlockNumber = block.number;
             } else {
-                delete stakes[overlay];
+                delete stakes[_owner];
             }
         }
-        emit StakeSlashed(overlay, amount);
+        emit StakeSlashed(_owner, stakes[_owner].overlay, _amount);
+    }
+
+    /**
+     * @dev Change overlay of address to new one for neighbourhood hopping
+     * @param _owner the _owner adress selected
+     * @param _nonce the new nonce that will produce overlay
+     */
+    function changeOverlay(address _owner, bytes32 _nonce) external {
```

Comment:
technically yes, that is kind of cool way to combine two and get results,
but then again we also have in depositStake **amount** so we have to add extra IF for overlay changes not to call token transfer on 0 and I am more leaning into separation of concerns and to have more readable code in this way. It would be a bit more gas spending on deployment which is fine and later it would be a little less gas for change overlay function and also we can have clean events for each action.